daffodil-propgen/src/test/scala/org/apache/daffodil/propGen/TestPropertyGenerator.scala

```diff
@@ -65,7 +65,7 @@ class TestPropertyGenerator {
     val mx = pg.genAttributeGroup(sch)
     assertTrue(mx.contains(""" = convertToBoolean(findProperty("prefixIncludesPrefixLength").value)"""))
     assertTrue(mx.contains("""LengthKindMixin"""))
-    assertTrue(mx.contains("""def lengthPropertiesAGInit() : Unit = {"""))
+    assertTrue(mx.contains("""def lengthPropertiesAGInit() (): Unit = {"""))
```

Comment:
Same issue here

Reply:
Yup. I'm counting on the CI to find the bugs for me.

Reply:
Looks likethe CI passed, I guess this is technically valid syntax since scala allows multiple parameter lists for currying, and I guess it has no problem with an empty extra parameter list?

Reply:
Strange. But yes you can 
`def foo(): Unit = { .... }`
then invoke with just
`foo` not `foo()`
And scala supplies the implied added '()'. So it seems it can do this even for
`def foo()(): Unit = { .... }`
which invokes with 
`foo()`, again supplying the second added '()' implicitly. 

Grrrr. Feels like a bug, not a feature. 

I'll chase down all ") ()". 

I've been thinking about Scala 2.13 and Scala 3.0 eventually, but unless we cross-support them (like we did for 2.11 and 2.12 for a time) so that 2.12 continues to be supported, we're stuck because other important Scala libraries like Apache Spark are still on 2.12.

In other words, I suspect we have to turn on these other warnings that we need for code-quality again without waiting for new scala versions. I.e., in Scala 2.12, and cope with them best we can. E.g., I think the warn on unused and warn on dead-code are both pretty important things for the compiler to do for us. 

A wildcard worthy of investigation is whether the Java modules stuff (introduced in Java 9 I believe, and which seems not to be the OGSI stuff all over again) allows one to link together modules with different dependencies, so long as the crossing between them is a a Java API.